From 92e18f08e3b94588628e314a63d9e78ebd922bf0 Mon Sep 17 00:00:00 2001
From: Wang Jie <dave.wang@rock-chips.com>
Date: Thu, 5 Sep 2019 16:23:18 +0800
Subject: [PATCH] input: sensors: program the sensor driver into the kernel.

Modify the code to sensor driver into the kernel and
make the compilation successful.

Change-Id: I3226b5df0084a80c63b7ad59aec7c759b3165a7b
Signed-off-by: Wang Jie <dave.wang@rock-chips.com>
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/input/Kconfig                          |  6 ++++--
 drivers/input/Makefile                         |  4 ++--
 drivers/input/sensors/accel/dmard10.c          |  2 +-
 drivers/input/sensors/accel/kxtik.c            |  2 +-
 drivers/input/sensors/accel/kxtj9.c            |  2 +-
 drivers/input/sensors/accel/lis3dh.c           |  2 +-
 drivers/input/sensors/accel/lsm303d.c          |  2 +-
 drivers/input/sensors/accel/mma7660.c          |  2 +-
 drivers/input/sensors/accel/mma8452.c          |  2 +-
 drivers/input/sensors/accel/mpu6500_acc.c      |  2 +-
 drivers/input/sensors/accel/mpu6880_acc.c      |  2 +-
 drivers/input/sensors/accel/mxc622x.c          |  2 +-
 drivers/input/sensors/compass/ak8975.c         | 17 +++--------------
 drivers/input/sensors/gyro/ewtsa.c             |  2 +-
 drivers/input/sensors/gyro/l3g20d.c            |  2 +-
 drivers/input/sensors/gyro/l3g4200d.c          |  2 +-
 drivers/input/sensors/gyro/mpu6500_gyro.c      |  2 +-
 drivers/input/sensors/gyro/mpu6880_gyro.c      |  2 +-
 drivers/input/sensors/hall/och165t_hall.c      |  2 +-
 drivers/input/sensors/lsensor/cm3217.c         |  2 +-
 drivers/input/sensors/lsensor/cm3232.c         |  2 +-
 drivers/input/sensors/lsensor/isl29023.c       |  2 +-
 drivers/input/sensors/lsensor/ls_al3006.c      |  2 +-
 drivers/input/sensors/lsensor/ls_ap321xx.c     |  2 +-
 drivers/input/sensors/lsensor/ls_stk3171.c     |  2 +-
 drivers/input/sensors/lsensor/ls_us5152.c      |  2 +-
 drivers/input/sensors/pressure/pr_ms5607.c     |  2 +-
 drivers/input/sensors/psensor/ps_al3006.c      |  2 +-
 drivers/input/sensors/psensor/ps_ap321xx.c     |  2 +-
 drivers/input/sensors/psensor/ps_stk3171.c     |  2 +-
 drivers/input/sensors/sensor-dev.c             |  6 +++---
 drivers/input/sensors/temperature/tmp_ms5607.c |  2 +-
 include/linux/sensor-dev.h                     |  1 +
 33 files changed, 41 insertions(+), 49 deletions(-)

diff --git a/drivers/input/Kconfig b/drivers/input/Kconfig
index 849d54c4c9da..77552ad5560b 100644
--- a/drivers/input/Kconfig
+++ b/drivers/input/Kconfig
@@ -196,10 +196,12 @@ source "drivers/input/tablet/Kconfig"
 
 source "drivers/input/touchscreen/Kconfig"
 
-source "drivers/input/misc/Kconfig"
-
 source "drivers/input/remotectl/Kconfig"
 
+source "drivers/input/sensors/Kconfig"
+
+source "drivers/input/misc/Kconfig"
+
 source "drivers/input/rmi4/Kconfig"
 
 endif
diff --git a/drivers/input/Makefile b/drivers/input/Makefile
index 268249852af6..fea58a026565 100644
--- a/drivers/input/Makefile
+++ b/drivers/input/Makefile
@@ -24,10 +24,10 @@ obj-$(CONFIG_INPUT_MOUSE)	+= mouse/
 obj-$(CONFIG_INPUT_JOYSTICK)	+= joystick/
 obj-$(CONFIG_INPUT_TABLET)	+= tablet/
 obj-$(CONFIG_INPUT_TOUCHSCREEN)	+= touchscreen/
+obj-$(CONFIG_ROCKCHIP_REMOTECTL)+= remotectl/
+obj-$(CONFIG_SENSOR_DEVICE)	+= sensors/
 obj-$(CONFIG_INPUT_MISC)	+= misc/
 
 obj-$(CONFIG_INPUT_APMPOWER)	+= apm-power.o
 
-obj-$(CONFIG_ROCKCHIP_REMOTECTL)	+= remotectl/
-
 obj-$(CONFIG_RMI4_CORE)		+= rmi4/
diff --git a/drivers/input/sensors/accel/dmard10.c b/drivers/input/sensors/accel/dmard10.c
index 6aeee09655de..eb92d1ba87b7 100644
--- a/drivers/input/sensors/accel/dmard10.c
+++ b/drivers/input/sensors/accel/dmard10.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/kxtik.c b/drivers/input/sensors/accel/kxtik.c
index 48276467df89..099fdaa69ca7 100644
--- a/drivers/input/sensors/accel/kxtik.c
+++ b/drivers/input/sensors/accel/kxtik.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/kxtj9.c b/drivers/input/sensors/accel/kxtj9.c
index b8cfeef57c27..f8aef8291a9b 100644
--- a/drivers/input/sensors/accel/kxtj9.c
+++ b/drivers/input/sensors/accel/kxtj9.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/lis3dh.c b/drivers/input/sensors/accel/lis3dh.c
index d0cc99e3effa..3dc3bb62d2e2 100644
--- a/drivers/input/sensors/accel/lis3dh.c
+++ b/drivers/input/sensors/accel/lis3dh.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/lsm303d.c b/drivers/input/sensors/accel/lsm303d.c
index 17a5baeb98b3..10a81a2fdc3e 100644
--- a/drivers/input/sensors/accel/lsm303d.c
+++ b/drivers/input/sensors/accel/lsm303d.c
@@ -20,7 +20,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/mma7660.c b/drivers/input/sensors/accel/mma7660.c
index 77dad66141cb..fb1e9faea6ab 100644
--- a/drivers/input/sensors/accel/mma7660.c
+++ b/drivers/input/sensors/accel/mma7660.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/mma8452.c b/drivers/input/sensors/accel/mma8452.c
index a061bbd886e6..3c8e29b24de2 100644
--- a/drivers/input/sensors/accel/mma8452.c
+++ b/drivers/input/sensors/accel/mma8452.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/mpu6500_acc.c b/drivers/input/sensors/accel/mpu6500_acc.c
index 042ddfe21a58..d355898805fa 100644
--- a/drivers/input/sensors/accel/mpu6500_acc.c
+++ b/drivers/input/sensors/accel/mpu6500_acc.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/mpu6880_acc.c b/drivers/input/sensors/accel/mpu6880_acc.c
index 105b42003734..218299ec7f56 100644
--- a/drivers/input/sensors/accel/mpu6880_acc.c
+++ b/drivers/input/sensors/accel/mpu6880_acc.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/accel/mxc622x.c b/drivers/input/sensors/accel/mxc622x.c
index 9b4e1b120dd0..a0439cc6c6ac 100644
--- a/drivers/input/sensors/accel/mxc622x.c
+++ b/drivers/input/sensors/accel/mxc622x.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/compass/ak8975.c b/drivers/input/sensors/compass/ak8975.c
index 8bbc1a0c64a9..8cec3b541252 100644
--- a/drivers/input/sensors/compass/ak8975.c
+++ b/drivers/input/sensors/compass/ak8975.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
@@ -183,7 +183,6 @@ static int sensor_report_value(struct i2c_client *client)
 	unsigned char *stat;
 	unsigned char *stat2;
 	int ret = 0;
-	char value = 0;
 	int i;
 
 	if(sensor->ops->read_len < 8)	//sensor->ops->read_len = 8
@@ -257,8 +256,8 @@ static int sensor_report_value(struct i2c_client *client)
 	if((sensor->pdata->irq_enable)&& (sensor->ops->int_status_reg >= 0))	//read sensor intterupt status register
 	{
 
-		value = sensor_read_reg(client, sensor->ops->int_status_reg);
-		DBG("%s:sensor int status :0x%x\n",__func__,value);
+		DBG("%s:sensor int status :0x%x\n", __func__,
+			sensor_read_reg(client, sensor->ops->int_status_reg));
 	}
 
 
@@ -430,7 +429,6 @@ static long compass_dev_ioctl(struct file *file,
 	struct i2c_client *client = this_client;
 	void __user *argp = (void __user *)arg;
 	int result = 0;
-	struct akm_platform_data compass;
 
 	/* NOTE: In this function the size of "char" should be 1-byte. */
 	char compass_data[SENSOR_DATA_SIZE];/* for GETDATA */
@@ -543,15 +541,6 @@ static long compass_dev_ioctl(struct file *file,
 			delay = sensor->flags.delay;
 			break;
 		case ECS_IOCTL_GET_PLATFORM_DATA:
-			DBG("%s:ECS_IOCTL_GET_PLATFORM_DATA start\n",__func__);
-			//memcpy(compass.m_layout, sensor->pdata->m_layout, sizeof(sensor->pdata->m_layout));
-			//memcpy(compass.project_name, sensor->pdata->project_name, sizeof(sensor->pdata->project_name));
-			ret = copy_to_user(argp, &compass, sizeof(compass));
-			if(ret < 0)
-			{
-				printk("%s:error,ret=%d\n",__FUNCTION__, ret);
-				return ret;
-			}
 			break;
 
 		default:
diff --git a/drivers/input/sensors/gyro/ewtsa.c b/drivers/input/sensors/gyro/ewtsa.c
index 66951c227203..4756ed24c0a7 100644
--- a/drivers/input/sensors/gyro/ewtsa.c
+++ b/drivers/input/sensors/gyro/ewtsa.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/gyro/l3g20d.c b/drivers/input/sensors/gyro/l3g20d.c
index 5f914168b7f5..300acd3f28ca 100644
--- a/drivers/input/sensors/gyro/l3g20d.c
+++ b/drivers/input/sensors/gyro/l3g20d.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/gyro/l3g4200d.c b/drivers/input/sensors/gyro/l3g4200d.c
index de1ff1d731b3..508c907b1905 100644
--- a/drivers/input/sensors/gyro/l3g4200d.c
+++ b/drivers/input/sensors/gyro/l3g4200d.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/gyro/mpu6500_gyro.c b/drivers/input/sensors/gyro/mpu6500_gyro.c
index 4464c20ffebb..fb18fff7cfcb 100644
--- a/drivers/input/sensors/gyro/mpu6500_gyro.c
+++ b/drivers/input/sensors/gyro/mpu6500_gyro.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/gyro/mpu6880_gyro.c b/drivers/input/sensors/gyro/mpu6880_gyro.c
index 883bc24b7d2e..9ff6a65819b0 100644
--- a/drivers/input/sensors/gyro/mpu6880_gyro.c
+++ b/drivers/input/sensors/gyro/mpu6880_gyro.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/hall/och165t_hall.c b/drivers/input/sensors/hall/och165t_hall.c
index c8547112b8f2..6318708380fb 100644
--- a/drivers/input/sensors/hall/och165t_hall.c
+++ b/drivers/input/sensors/hall/och165t_hall.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/lsensor/cm3217.c b/drivers/input/sensors/lsensor/cm3217.c
index b23cb77cfb7a..7694d5c85416 100755
--- a/drivers/input/sensors/lsensor/cm3217.c
+++ b/drivers/input/sensors/lsensor/cm3217.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/lsensor/cm3232.c b/drivers/input/sensors/lsensor/cm3232.c
index 2afd1d79e82d..f0084f890fcd 100755
--- a/drivers/input/sensors/lsensor/cm3232.c
+++ b/drivers/input/sensors/lsensor/cm3232.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/lsensor/isl29023.c b/drivers/input/sensors/lsensor/isl29023.c
index cd649e4a4ff2..2ed0a41071a6 100755
--- a/drivers/input/sensors/lsensor/isl29023.c
+++ b/drivers/input/sensors/lsensor/isl29023.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/lsensor/ls_al3006.c b/drivers/input/sensors/lsensor/ls_al3006.c
index 05b4c233d6b0..74beccf07ee5 100755
--- a/drivers/input/sensors/lsensor/ls_al3006.c
+++ b/drivers/input/sensors/lsensor/ls_al3006.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/lsensor/ls_ap321xx.c b/drivers/input/sensors/lsensor/ls_ap321xx.c
index 225d7a4d9829..597f19dc41b8 100644
--- a/drivers/input/sensors/lsensor/ls_ap321xx.c
+++ b/drivers/input/sensors/lsensor/ls_ap321xx.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/lsensor/ls_stk3171.c b/drivers/input/sensors/lsensor/ls_stk3171.c
index e50e9c632842..bf71351deabb 100755
--- a/drivers/input/sensors/lsensor/ls_stk3171.c
+++ b/drivers/input/sensors/lsensor/ls_stk3171.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/lsensor/ls_us5152.c b/drivers/input/sensors/lsensor/ls_us5152.c
index 93f09284b73d..1d041c32bd1b 100644
--- a/drivers/input/sensors/lsensor/ls_us5152.c
+++ b/drivers/input/sensors/lsensor/ls_us5152.c
@@ -5,7 +5,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/pressure/pr_ms5607.c b/drivers/input/sensors/pressure/pr_ms5607.c
index 7690c14cccc5..65a53af8e2ba 100755
--- a/drivers/input/sensors/pressure/pr_ms5607.c
+++ b/drivers/input/sensors/pressure/pr_ms5607.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/psensor/ps_al3006.c b/drivers/input/sensors/psensor/ps_al3006.c
index 000af717ac8d..c9ef00d8b06b 100755
--- a/drivers/input/sensors/psensor/ps_al3006.c
+++ b/drivers/input/sensors/psensor/ps_al3006.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/psensor/ps_ap321xx.c b/drivers/input/sensors/psensor/ps_ap321xx.c
index 8a656c01814f..606fbb8989b8 100644
--- a/drivers/input/sensors/psensor/ps_ap321xx.c
+++ b/drivers/input/sensors/psensor/ps_ap321xx.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/psensor/ps_stk3171.c b/drivers/input/sensors/psensor/ps_stk3171.c
index 53471a4c5579..03a918b0d526 100755
--- a/drivers/input/sensors/psensor/ps_stk3171.c
+++ b/drivers/input/sensors/psensor/ps_stk3171.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/drivers/input/sensors/sensor-dev.c b/drivers/input/sensors/sensor-dev.c
index 99b36c04f97f..f78493319b58 100644
--- a/drivers/input/sensors/sensor-dev.c
+++ b/drivers/input/sensors/sensor-dev.c
@@ -20,7 +20,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
@@ -241,7 +241,7 @@ static ssize_t accel_calibration_store(struct class *class,
 	return ret ? ret : count;
 }
 
-static CLASS_ATTR(accel_calibration, 0664, accel_calibration_show, accel_calibration_store);
+static CLASS_ATTR_RW(accel_calibration);
 
 static ssize_t gyro_calibration_show(struct class *class,
 		struct class_attribute *attr, char *buf)
@@ -367,7 +367,7 @@ static ssize_t gyro_calibration_store(struct class *class,
 	return ret ? ret : count;
 }
 
-static CLASS_ATTR(gyro_calibration, 0664, gyro_calibration_show, gyro_calibration_store);
+static CLASS_ATTR_RW(gyro_calibration);
 
 static int sensor_class_init(void)
 {
diff --git a/drivers/input/sensors/temperature/tmp_ms5607.c b/drivers/input/sensors/temperature/tmp_ms5607.c
index aea422286ba7..8855ada3f2e8 100755
--- a/drivers/input/sensors/temperature/tmp_ms5607.c
+++ b/drivers/input/sensors/temperature/tmp_ms5607.c
@@ -19,7 +19,7 @@
 #include <linux/irq.h>
 #include <linux/miscdevice.h>
 #include <linux/gpio.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/atomic.h>
 #include <linux/delay.h>
 #include <linux/input.h>
diff --git a/include/linux/sensor-dev.h b/include/linux/sensor-dev.h
index 19a8f03f3e0c..4881c9ed0a50 100644
--- a/include/linux/sensor-dev.h
+++ b/include/linux/sensor-dev.h
@@ -20,6 +20,7 @@
 #endif
 
 #include <dt-bindings/sensor-dev.h>
+#include <linux/module.h>
 
 #define SENSOR_ON		1
 #define SENSOR_OFF		0
-- 
2.35.3

